addons/purchase_stock/models/stock.py

```diff
@@ -216,6 +217,27 @@ def action_view_purchase(self):
 
         return result
 
+    def action_replenish(self):
+        if self.route_id == self.env.ref('purchase_stock.route_warehouse0_buy'):
```

Comment:
Too specific. User could create it's own route with buy action. Intsead check the route action and if once contains action == 'buy' then do that